File: BFS.py
def adj_list(n, edges):
    graph = [[] for i in range(n)]
    for e in edges:
        a = e[0]
        b = e[1]
        graph[a].append(b)
        graph[b].append(a)
    return graph


class Queue:
    def __init__(self):
        '''Make a new empty queue'''
        self.q = []

    def enqueue(self, o):
        '''add o to the end of the queue'''
        self.q.append(o)

    def dequeue(self):
        '''remove the front element from the queue'''
        return self.q.pop(0)

    def front(self):
        '''return the front element from the queue'''
        return self.q[0]

    def isEmpty(self):
        return len(self.q) == 0

    def size(self):
        return len(self.q)


def bfs(graph, visited):
    q = Queue()
    q.enqueue(0)
    visited[0] = True
    while not q.isEmpty():
        current_node = q.front()
        print(current_node)
        q.dequeue()
        for node in graph[current_node]:
            if visited[node]:
                continue
            q.enqueue(node)
            visited[node] = True


n = 5
edges = [(0, 1),(0, 2),(0, 4),(1, 2),(1, 3),(2, 3)]

graph = adj_list(n, edges)
visited = [False for i in range(n)]

bfs(graph, visited)

File: BFS.java
import java.util.LinkedList;

public class BFS {

    public static LinkedList<Integer>[] adj_list(int n, int[][] edges) {
        LinkedList<Integer>[] graph = new LinkedList[n];
        for(int i = 0; i < n; i++) {
            graph[i] = new LinkedList<>();
        }

        for(int[] e : edges) {
            int a = e[0];
            int b = e[1];
            graph[a].add(b);
            graph[b].add(a);
        }

        return graph;
    }

    public static void bfs(LinkedList<Integer>[] graph, boolean[] visited) {
        Queue q = new Queue();
        q.enqueue(0);
        visited[0] = true;

        while(!q.isEmpty()) {
            int current_node = q.front();
            System.out.println(current_node);
            q.dequeue();
            int val;
            for(int i = 0; i < graph[current_node].size(); i++) {
                val = graph[current_node].get(i);
                if(visited[val] == false) {
                    q.enqueue(val);
                    visited[val] = true;
                }
            }
        }
    }

    public static void main(String[] args) {
        int n = 5;
        int[][] edges = {{0, 1}, {0, 2}, {0, 4}, {1, 2}, {1, 3}, {2, 3}};
        LinkedList<Integer>[] graph = adj_list(n, edges);
        
        boolean[] visited = new boolean[n];
        for(int i = 0; i < n; i++) {
         visited[i] = false;
        }
 
        bfs(graph, visited);
    } 
}